client/verta/tests/conftest.py

```diff
@@ -356,14 +356,17 @@ def commit(repository):
 
 
 @pytest.fixture
-def created_datasets(client):
+def created_datasets(client, client_2):
     """Container to track and clean up Datasets created during tests."""
     created_datasets = []
 
     yield created_datasets
 
     if created_datasets:
-        utils.delete_datasets(list(set(dataset.id for dataset in created_datasets)), client._conn)
+        try:
+            utils.delete_datasets(list(set(dataset.id for dataset in created_datasets)), client._conn)
+        except requests.HTTPError:
+            utils.delete_datasets(list(set(dataset.id for dataset in created_datasets)), client_2._conn)
```

Comment:
This wouldn't work either, because the `dataset._conn` is outside the context of the list comprehension which defines dataset.

I think what I just pushed should work.